astropy/table/table.py

```diff
@@ -2165,6 +2157,27 @@ def rename_column(self, name, new_name):
 
         self.columns[name].info.name = new_name
 
+    def _set_row(self, idx, colnames, vals):
+        try:
```

Comment:
It is probably overkill, but we could be even more agnostic about `value` by replacing this with, just after the loop and before the exception:
```
# Check that we got to the end of names (which, unlike vals,
# is known to be a list of tuple of str, so indexable).
if name is not names[-1]:
    raise ValueError('right ...')
```
It should even be slightly faster... But perhaps not as clear (and clarity does matter).

Reply:
I don't think that catches the case where `vals` is an iterable that is *longer* than `colnames` (where `zip` cuts off because of `colnames`).

Reply:
true, well, that decides it I think!